corehq/apps/reports/views.py

```diff
@@ -1756,7 +1756,11 @@ def _error(msg):
         instance = _get_location_safe_form(domain, request.couch_user, instance_id)
         context = _get_form_context(request, domain, instance)
         if not instance.app_id or not instance.build_id:
-            return _error(_('Could not detect the application/form for this submission.'))
+            return _error(_(
+                "This submission was made from the App Preview."
+                "As a result it is not possible to edit this submission\n\n"
+                "If this was not submitted from App Preview, please report an issue"
```

Comment:
Hmm. that's a good point. I didn't mean to push uncertainty to the users, just meant to give a more precie message.

We could also assert that the deviceID is Formplayer and/or it was submitted by a webuser. If not, we could show the previous error.